plugins/main/public/components/common/modules/main-agent.tsx

```diff
@@ -66,179 +73,124 @@ export class MainModuleAgent extends Component {
 
   renderTitle() {
     return (
-      <EuiFlexGroup>
-        <EuiFlexItem className='wz-module-header-agent-title'>
-          <EuiFlexGroup>
-            <EuiFlexItem grow={false}>
-              <span style={{ display: 'inline-flex' }}>
-                <EuiTitle size='s' className='wz-module-header-agent-title-btn'>
-                  <h1>
-                    <span
-                      style={{ color: euiThemeVars.euiColorPrimaryText }}
-                      onClick={() => {
-                        NavigationService.getInstance().navigate(
-                          `/agents?tab=welcome&agent=${this.props.agent.id}`,
-                        );
-                      }}
-                    >
-                      <span>
-                        &nbsp;{this.props.agent.name}&nbsp;&nbsp;&nbsp;
-                      </span>
-                    </span>
-                  </h1>
-                </EuiTitle>
-              </span>
-            </EuiFlexItem>
-            <EuiFlexItem />
-            {this.props.section === 'syscollector' && (
-              <EuiFlexItem
-                grow={false}
-                style={{ marginRight: 4, marginTop: 6 }}
-              >
-                <GenerateSyscollectorReportButton agent={this.props.agent} />
-              </EuiFlexItem>
-            )}
-          </EuiFlexGroup>
+      <EuiFlexGroup style={{ marginInline: 8 }}>
+        <EuiFlexItem style={{ marginInline: 0 }} grow={false}>
+          {this.state.loadingReport ? (
+            <EuiLoadingSpinner size='s' />
+          ) : (
+            <EuiTabs>
+              {[AgentTabs.SOFTWARE, AgentTabs.NETWORK, AgentTabs.PROCESSES].includes(this.props.section) ? (
+                <>
+                  <EuiTab
+                    isSelected={this.props.section === AgentTabs.SOFTWARE}
+                    onClick={() => this.props.switchTab?.(AgentTabs.SOFTWARE)}
+                  >
+                    {toTitleCase(AgentTabs.SOFTWARE)}&nbsp;
+                  </EuiTab>
+                  <EuiTab
+                    isSelected={this.props.section === AgentTabs.NETWORK}
+                    onClick={() => this.props.switchTab?.(AgentTabs.NETWORK)}
+                  >
+                    {toTitleCase(AgentTabs.NETWORK)}&nbsp;
+                  </EuiTab>
+                  <EuiTab
+                    isSelected={this.props.section === AgentTabs.PROCESSES}
+                    onClick={() => this.props.switchTab?.(AgentTabs.PROCESSES)}
+                  >
+                    {toTitleCase(AgentTabs.PROCESSES)}&nbsp;
+                  </EuiTab>
```

Comment:
**nitpick**: this could be refactored using and array and mapping the render.

Reply:
Fixed